### PR TITLE
Add rts_flag to JPL Horizons ephemerides

### LINKED ISSUE #1799: RTS ONLY for JPL Horizons
Will be great if the JPL Horizons Ephemerides could deal with RTS (rise-transit-set only)

The CGI uses the `rts_flag` property, but the batch processor seems to need the `R_T_S_ONLY` param.

More details at https://ssd.jpl.nasa.gov/?horizons_doc#rts

### PR DESCRIPTION
**Pull request by XaviTorello**
It ensures to output data only at target RTS (rise-transit-set) passing the needed `RTS ONLY` flag to the CGI

It provides the new `rts_flag` optional parameter for the Horizons.ephemeris method

See https://ssd.jpl.nasa.gov/?horizons_doc#rts for more information

If activated, returns:

```
# example with `rts_flag = "TVH`  //true visual horizon plane

targetname    datetime_str      datetime_jd    solar_presence flags      AZ         EL    
   ---            ---                d              ---        ---      deg        deg    
---------- ----------------- ----------------- -------------- ----- ----------- ----------
  Sun (10) 2020-Aug-20 05:03 2459081.710416667              *     r  71.8343523 -0.7719307
  Sun (10) 2020-Aug-20 11:57 2459081.997916667              *     t 179.9287244 59.6468702
  Sun (10) 2020-Aug-20 18:51 2459082.285416667              C     s 287.9787747 -0.8537235
```

### Full example

```python
from astroquery.jplhorizons import Horizons
from astropy.time import Time

time = ["2020-08-20T00:00:00"]
t = Time(time, format="isot", scale="utc")

ordino = {"lon": 1.532610, "lat": 42.556761, "elevation": 0.3}

horizons_params = {
    "id_type": "majorbody",
    "location": ordino,
    "epochs": {
        "start": t.to_datetime()[0].__str__(),
        "stop": (t + 1).to_datetime()[0].__str__(),
        "step": "1m",
    },
}
ephemerides_params = {
    "quantities": "4",
    "refraction": True,
    "rts_flag": "TVH",
}

sun = Horizons(id=10, **horizons_params,)
ephemerides = sun.ephemerides(**ephemerides_params)
```

Fix #1799

### CODE REVIEW

**Comment by XaviTorello** [comment]
What do you think @bsipocz, can it be merged? Ty!

File: astroquery/jplhorizons/core.py [hunk 1]
```diff
@@ -445,6 +446,11 @@ def ephemerides_async(self, airmass_lessthan=99,
             Observer Table Quantities
             <https://ssd.jpl.nasa.gov/?horizons_doc#table_quantities>`_;
             default: all quantities
+        rts_flag: string, optional
+            Output data only at target rise/transit/set (RTS)
+            `RTS only print mode
+            <https://ssd.jpl.nasa.gov/?horizons_doc#rts>`_;
+            default: disabled
```
Comment (migueldvb) [inline astroquery/jplhorizons/core.py:453]: In the docstring you could mention the accepted values for `rts_flag` (TVH, GEO, RAD) with a brief description to make the documentation as self-contained as possible.

File: astroquery/jplhorizons/core.py [hunk 2]
```diff
@@ -584,6 +590,9 @@ def ephemerides_async(self, airmass_lessthan=99,
         else:
             request_payload['SKIP_DAYLT'] = 'NO'
 
+        if rts_flag:
+            request_payload['R_T_S_ONLY'] = rts_flag
+
```
Comment (migueldvb) [inline astroquery/jplhorizons/core.py:595]: In addition it is helpful to check the parameter and provide an informative error message.

```python
if rts_flag and rts_flag not in ('TVH', 'GEO', 'RAD'):
    raise ValueError("Invalid rts_flag specified. Valid types are TVH, GEO and RAD")
```

Comment (XaviTorello) [reply astroquery/jplhorizons/core.py:595]: Not sure if it's a good idea if we hardcode current valid options.

As I understand this library just should be a "transparent" proxy of the end-service backend. A link to the service settings is provided in the docstring to be able to quickly identify the accepted options (to JPL docs).

I mean, if the backend changes (JPL in this case) a new PR should be raised to support new settings; if we keep this as simple as possible delegating the payloads validation to the backed we'll ensure that no downtime nor updates are needed to still support new functionalities.

It plays with the same idea that I propose in #1802, pass the payload to the backend delegating the validation and just raise if errors appears.

But if you think that this assertions should be performed I can change it without any effort :)

Thanks @migueldvb

File: astroquery/jplhorizons/core.py (hunk 1 above)
Comment (XaviTorello) [reply astroquery/jplhorizons/core.py:453]: See my other comment

File: astroquery/jplhorizons/core.py (hunk 2 above)
Comment (migueldvb) [reply astroquery/jplhorizons/core.py:595]: That sounds good to me. I think that his can be merged as it is if there are no conflicts with the master branch.

Comment (XaviTorello) [reply astroquery/jplhorizons/core.py:595]: Perfect!

**Comment by XaviTorello** [comment]
All conversations has been resolved, seems ready to be merged

File: astroquery/jplhorizons/core.py [hunk 3]
```diff
@@ -149,6 +149,7 @@ def ephemerides_async(self, airmass_lessthan=99,
                           refsystem='J2000',
                           closest_apparition=False, no_fragments=False,
                           quantities=conf.eph_quantities,
+                          rts_flag=None,
```
Comment (bsipocz) [inline astroquery/jplhorizons/core.py:152]: this breaks API. Either add to the end of kwargs, or make kwargs keyword only by inserting a `*` after the mandatory parameters.

**Review by bsipocz — CHANGES_REQUESTED** [review]
This introduces an API break which should be resolved. And also, please add a changelog entry that mentions the new keyword.

**Comment by bsipocz** [comment]
with #1802 merged, this is now outdated as `R_T_S_ONLY` can now be passed on via the `optional settings` kwarg.

Please do reopen if you disagree and rather see another kwarg specifically for this `rts` setting.

**Comment by bsipocz** [comment]
Has been addressed by #1802

**Pull request closed without merging**